### ISSUE TITLE
Improve 1D feature protection API

### ISSUE THREAD

**Issue by sloriot** [CGAL/cgal#5171]
Currently there is no easy way to see what the 1D protection algorithm is doing and when it fails (i.e. over-refinement)

An API should be added to run the protection algorithm (doing for example an initialization of `c3t3.triangulation()`) and that function should return how the protection when.  Being able to specifying the minimal size of the ball protection would be nice as it could help now about over-refinement.

Being able to visualize balls would also be interesting, and also ball "in conflict/invalid".

**Comment by sloriot** [comment]
At the same time, it would be nice to expose the hidden parameters to limit the size of the minimal protecting ball. Having some error code returned would simplify the debugging.

**Comment by sloriot** [comment]
@janetournois this is partially implemented with min_edge_size, right?

**Comment by janetournois** [comment]
Indeed, `edge_min_size` prevents over-refinement by bounding the size of protecting balls from below.

However, there is no return code to know if feature protection has been stopped by this criterion or if it has terminated "normally"
